src/about/settings/index.js

```diff
@@ -28,7 +55,99 @@
 
   // View
 
-  const viewSetting = (name, value) => html.div({
+  const viewField = props => {
+    return html.span({
+      key: 'field',
+      style: {
+        position: 'relative',
+        overflow: 'hidden'
+      }
+    },[
+      html.span({
+        key: 'sizer',
+        style: {
+          left: 0,
+          color: 'red',
+          whiteSpace: 'pre',
+          opacity: 0,
+        }
+      }, props.value),
+      input({
+        style: {
+          padding: 0,
+          margin: 0,
+          background: 'transparent',
+          border: 'none',
+          font: 'inherit',
+          margin: 0,
+          padding: 0,
+          position: 'absolute',
+          zIndex: 2,
+          left: 0,
+          width: '100%'
+        },
+        type: 'text',
+        text: props.value,
+        onChange: props.onChange
+      })
+    ]);
+  };
+
+  const viewValue = (name, value, address) => {
+    const type = value == null ? 'null' :
+          typeof(value);
+
+    return type === 'string' ? viewString(name, value, address) :
+           type === 'number' ? viewNumber(name, value, address) :
+           type === 'boolean' ? viewBoolean(name, value, address) :
+           viewJSON(name, value, address);
+  }
+
+  const viewNumber = (name, value, address) =>
+    html.input({
+      type: 'number',
+      style: {
+        border: 'none',
+        width: 'auto'
+      },
+      value,
+      onChange: event => address.receive(Settings.Update({
+        name, value: event.target.valueAsNumber
+      }))});
+
+  const viewBoolean = (name, value, address) =>
+    html.input({
+      type: 'checkbox',
+      checked: value,
+      onChange: event => address.receive(Settings.Update({
+        name, value: event.target.checked
+      }))});
+
+  const viewString = (name, value, address) =>
+    html.code(null, [
+      '"',
```

Comment:
Well you still may change the type if you say edit `null` field to `"null"` it will become `"null"` instead. But anyway maybe you're right and I did over engineered this.